src/jit/codegenxarch.cpp

```diff
@@ -4715,8 +4725,24 @@ instruction CodeGen::genGetInsForOper(genTreeOps oper, var_types type)
 void CodeGen::genCodeForShift(GenTreePtr tree)
 {
     // Only the non-RMW case here.
+#ifdef _TARGET_64BIT_
     assert(tree->OperIsShiftOrRotate());
-    assert(!tree->gtOp.gtOp1->isContained());
+#else
+    assert(tree->OperIsShiftOrRotate() || tree->OperGet() == GT_LSH_HI || tree->OperGet() == GT_RSH_LO);
+#endif
+
```

Comment:
Given the number of `#ifdefs` in that you've added, and the lack of commonality with the new code and the old, how about adding instead a new genCodeForShiftLong() function that just handles the two new special HI opcodes?